cmd/eval.go

```diff
@@ -220,31 +236,38 @@ func determineInputSpec(arguments args) (*appstudioshared.ApplicationSnapshotSpe
 }
 
 func validateImage(ctx context.Context, imageRef, policyConfiguration, publicKey, rekorURL string) (*policy.Output, error) {
-	out := &policy.Output{}
+	imageValidator, err := image.NewImageValidator(ctx, imageRef, publicKey, rekorURL)
+	if err != nil {
+		return nil, err
+	}
 
-	i, err := image.NewImageValidator(ctx, imageRef, publicKey, rekorURL)
+	policyEvaluator, err := policy.NewPolicyEvaluator(policyConfiguration)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := i.ValidateImageSignature(ctx); err != nil {
+	return validateImageWith(ctx, imageRef, policyConfiguration, publicKey, rekorURL, imageValidator, policyEvaluator)
+}
+
+func validateImageWith(ctx context.Context, imageRef, policyConfiguration, publicKey, rekorURL string, imageValidator image.ImageValidator, policyEvaluator policy.PolicyEvaluator) (*policy.Output, error) {
+	out := &policy.Output{}
+	if err := imageValidator.ValidateImageSignature(ctx); err != nil {
 		out.SetImageSignatureCheck(false, err.Error())
-		return nil, err
+	} else {
+		out.SetImageSignatureCheck(true, "success")
 	}
-	out.SetImageSignatureCheck(true, "success")
 
-	if err := i.ValidateAttestationSignature(ctx); err != nil {
+	if err := imageValidator.ValidateAttestationSignature(ctx); err != nil {
 		out.SetAttestationSignatureCheck(false, err.Error())
-		return nil, err
+	} else {
+		out.SetAttestationSignatureCheck(true, "success")
 	}
-	out.SetAttestationSignatureCheck(true, "success")
 
-	p, err := policy.NewPolicyEvaluator(policyConfiguration)
-	if err != nil {
-		return nil, err
+	if !out.ImageSignatureCheck.Passed || !out.AttestationSignatureCheck.Passed {
+		return out, nil
 	}
 
-	results, err := p.Evaluate(ctx, i.Attestations())
+	results, err := policyEvaluator.Evaluate(ctx, imageValidator.Attestations())
 	if err != nil {
 		return nil, err
```

Comment:
Maybe this should call something like `out.SetPolicyCheck(false, err)` like the other checks.

Reply:
`policy.Output` holds [results](https://github.com/hacbs-contract/ec-cli/blob/70dbabae239e5f1f3ff2b3ef6582c0a604814c80/internal/policy/output.go#L35) from Conftest unlike image and attestation signature validations which hold [the flag and message](https://github.com/hacbs-contract/ec-cli/blob/70dbabae239e5f1f3ff2b3ef6582c0a604814c80/internal/policy/output.go#L33-L34) in `VerificationStatus`. Something we'll surely revisit as we're loosing quite a bit of fidelity from the policy output with this.